### PR TITLE
fix: visitArguments should travel sub fields

### PR DESCRIPTION
**Pull request by yjhmelody**
I meet some problems when do some transforms fro function call. And then I find the visitArguments have not done anything.

So I fix the base.ts, but baseTransform.ts cannot be fixed easily without api break change

### CODE REVIEW

File: src/base.ts [hunk 1]
```diff
@@ -398,11 +398,16 @@ export class BaseVisitor extends AbstractVisitor<Node> {
     this.visitArguments(node.typeArguments, node.args);
   }
 
-  visitArguments(
-    typeArguments: TypeNode[] | null,
+  visitArguments(typeArguments: TypeNode[] | null, args: Expression[]): void {
+    let typeArgs = typeArguments ? typeArguments : [];
+    for(const typeArg of typeArgs) {
```
Comment (willemneal) [inline src/base.ts:403]: Why not `this.visit(typeArgs)`?

Comment (yjhmelody) [reply src/base.ts:403]: The visit sig is `visit(node: T | null): void`

Comment (willemneal) [reply src/base.ts:403]: This signature?
https://github.com/willemneal/visitor-as/blob/057a447baee057fff88c432fb06050298ab08050/src/visitor.ts#L17

Comment (yjhmelody) [reply src/base.ts:403]: You are right

Comment (yjhmelody) [reply src/base.ts:403]: Should I also fix baseTransformer? But this leads to API breaking

**Pull request merged**